### ISSUE TITLE
Support for remote Docker hosts

### ISSUE THREAD

**Issue by zaps** [kitematic/kitematic#117]
I have a separate box running Docker daemon which I use OS X client to connect to. 
As such, installing VBox, boot2docker, etc is not necessary in this case.
What would be great is if this could be bypassed and just point the app to the Docker host.

**Comment by faddat** [comment]
Hear hear!  Is there any way that I could easily add this functionality to your code, dear authors?  It *sounds* like it should not be too difficult, but of course, I could be mistaken.  

The functionality described by Zaps was *exactly* what I wanted to use kitematic for, actually....

**Comment by dtornow** [comment]
+1

**Comment by DieterReuter** [comment]
Big :+1:
I'd like to see to support maybe more than a single remote Docker host!

Need help? Happy to help for sure.

**Comment by nickplee** [comment]
+1

**Comment by DieterReuter** [comment]
@jeffdm I just implemented an easy way to connect an external or remote Docker Host. Right now it works fine for me with my existing Boot2Docker VM. But there are some drawbacks with the function call of `docker-machine tls-regenerate-certs -f dev`, so I had to disable it - but I think this part should be included in a new upstream version of `docker-machine`.  And I did used my own version of `docker-machine` and had implemented a special driver to connect an existent Docker Host, this works great now on my Raspberry Pi running Docker 1.5.0.
Everything is included here in my PR https://github.com/hypriot/kitematic/pull/1
Disclamer: that's more a quick hack, but you can see it's easy to do.

**Comment by ev-tt** [comment]
+1 Have separate machine in my home setup for testing VMs and containers, would be nie to have this feature.
I see that @DieterReuter PR was more than two month ago. I'm wondering if it is going to be accepted soon or not?

**Comment by DieterReuter** [comment]
@ev-tt, I had done more a quick hack and this will not work the same way with the current Kitematic version. 

But we should work on it to get a more general solution to use different machine types. I'm also interested to support a couple machine at the same time, let's say control all the machines you have defined in docker machine - a local dev machine a a few cloud machines.

**Comment by christoth** [comment]
+1, I also have a separate server at home, that I'd love to use Kitematic

**Comment by DieterReuter** [comment]
@christoth, @ev-tt, we had a small conversation at Docker yesterday about adding this feature in Kitematic. I've done the work to support it with the current version too, but it needs to be refactored to get a great integration into upstream. I'd like to see a general integration with different Docker Engines, not only for a Raspi. I think we'll work this out soon after the DockerCon, so please stay tuned...

**Comment by ev-tt** [comment]
@DieterReuter thanks! Keep us updated!

**Comment by hit0wnzzz** [comment]
+1

**Comment by psychemedia** [comment]
So in general terms, would it be possible to support plugins for different cloud host providers, so I could eg create profiles with Digital Ocean or AWS or whoever (along with a default corresponding to the Kitematic default docker host) and then select the appropriate profile when I create the container (or just drag a hub image onto whatever profile context I want to launch that container in?)?

**Comment by christoth** [comment]
@DieterReuter thanks!

**Comment by rummand** [comment]
+1

**Comment by mohamnag** [comment]
external docker engine is definitely needed if this tool is more than a toy!

**Comment by mchiang0610** [comment]
We are working on this openly with contributors. Please feel free to voice your opinion as this is still in early draft: 

https://github.com/kitematic/kitematic/wiki/Machine-Management

**Comment by jeffdm** [comment]
Closing in favor of #231 to keep things in one place, but thanks for all the input!

**Comment by wrathagom** [comment]
This was closed, but the request that you referenced when doing so seems to focus around different VM software. I thought this request was more about remote management, as in docker-machine running on a different box and using the network to send commands to that box. I do most of my work on the new 12" macbook with an intel Core M. I'd rather have the VM running on my high end dev server.

That also seems easier to implement as it doesn't require waiting on docker machine to make any changes. The only thing that is required is to allow an override for where kinematic is connecting.

**Comment by FrenchBen** [comment]
@wrathagom The logic is still very similar, where Kitematic needs to have the ability to connect to 'another' machine than the 'default' it currently uses. 
All machines, local or remote, should be listed in `docker-machine` thus allowing Kitematic to work with them.
